src/test/java/org/littleshoot/proxy/DirectRequestTest.java

```diff
@@ -0,0 +1,163 @@
+            @Override
+            public HttpFilters filterRequest(HttpRequest originalRequest) {
+                return new HttpFiltersAdapter(originalRequest);
+            }
+        };
+
+        proxyServer = DefaultHttpProxyServer.bootstrap()//
+                .withPort(0)//
+                .withFiltersSource(filtersSource)//
+                .start();
+
+        int proxyPort = proxyServer.getListenAddress().getPort();
+        org.apache.http.HttpResponse response = getResponse("http://127.0.0.1:" + proxyPort + "/directToProxy");
+        int statusCode = response.getStatusLine().getStatusCode();
+
+        assertEquals("Expected to receive an HTTP 400 from the server", 400, statusCode);
+    }
+
+    @Test(timeout = 5000)
+    public void testAnswerFromFilterShouldBeServed() throws Exception {
+
+        HttpFiltersSource filtersSource = new HttpFiltersSourceAdapter() {
+            @Override
+            public HttpFilters filterRequest(HttpRequest originalRequest) {
+                return new HttpFiltersAdapter(originalRequest) {
+                    @Override
+                    public HttpResponse clientToProxyRequest(HttpObject httpObject) {
+                        return new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.valueOf(403));
+                    }
+                };
+            }
+        };
+
+        proxyServer = DefaultHttpProxyServer.bootstrap()//
+                .withPort(0)//
+                .withFiltersSource(filtersSource)//
+                .start();
+
+        int proxyPort = proxyServer.getListenAddress().getPort();
+        org.apache.http.HttpResponse response = getResponse("http://localhost:" + proxyPort + "/directToProxy");
+        int statusCode = response.getStatusLine().getStatusCode();
+
+        assertEquals("Expected to receive an HTTP 403 from the server", 403, statusCode);
+    }
+
+    @Test(timeout = 5000, expected = SSLException.class)
+    public void testHttpsShouldCancelConnection() throws Exception {
+
+        HttpFiltersSource filtersSource = new HttpFiltersSourceAdapter() {
+            @Override
+            public HttpFilters filterRequest(HttpRequest originalRequest) {
+                return new HttpFiltersAdapter(originalRequest) {
+                    @Override
+                    public HttpResponse clientToProxyRequest(HttpObject httpObject) {
+                        return new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.valueOf(403));
+                    }
+                };
+            }
+        };
+
+        proxyServer = DefaultHttpProxyServer.bootstrap()//
+                .withPort(0)//
+                .withFiltersSource(filtersSource)//
+                .start();
+
+        int proxyPort = proxyServer.getListenAddress().getPort();
+        getResponse("https://localhost:" + proxyPort + "/directToProxy");
+    }
+
+    private DefaultHttpClient buildHttpClient() throws Exception {
```

Comment:
No it was hand written. I've extracted it into a method for now. The next lines regarding the Apache client are copy/paste from ```AbstractProxyTest#buildHttpClient()```, since the implementation in ```HttpFilterTest#getDefaultHttpClient``` wont work for me. With this test I need no server and no proxy via, only a proxy and a client to get it direct.

It would be great to have an utility class, to create server, proxy and client, if we need it. I'll start with this separately.